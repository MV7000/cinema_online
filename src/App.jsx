import React from "react";
import { useState } from "react";

import { RouterProvider } from "react-router-dom";
import Context from "./Elements/Context/Context";

import router from "./router";
function App() {
  // Прописуємо основні стейти проекта
  const [stars, setStars] = useState([]),
    [search, setSearch] = useState([]),
    [films, setFilms] = useState([]),
    [closeModalTicket, setCloseModalTicket] = useState(false),
    [clozeHall, setClozeHall] = useState(true),
    [ticket, setTicket] = useState([]),
    [checkBuyTicket, setCheckBuyTicket] = useState(true),
    [myArr, setMyArr] = useState([]);

  // Функція рандомного обрання фільмів для слайдера головної сторінки
  const randomArr = [];
  if (films.results) {
    const randomIndex = (max) => Math.floor(Math.random() * Math.floor(max));
    for (let i = 0; i < 4; i++) {
      const index = randomIndex(20);
      randomArr.push(films.results[index]);
    }
  }
  // Дані для контексту проекта
  const methodsData = {
    stars,
    films,
    setStars,
    setFilms,
    randomArr,
    closeModalTicket,
    setCloseModalTicket,
    clozeHall,
    setClozeHall,
    ticket,
    setTicket,
    checkBuyTicket,
    setCheckBuyTicket,
    search,
    setSearch,
    myArr,
    setMyArr,
  };

  return (
    <Context.Provider value={methodsData}>
      <RouterProvider router={router}></RouterProvider>
    </Context.Provider>
  );
}

export default App;
