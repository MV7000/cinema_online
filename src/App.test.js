import { render, screen } from "@testing-library/react";
import App from "./App";
// Тести елементів проекту
describe("Elements on Main Page", () => {
  test("renders headers of galeries", () => {
    render(<App />);
    const moviesTitle = screen.getByText(/trending movies/i);
    const starsTitle = screen.getByText(/trending stars/i);
    expect(moviesTitle).toBeInTheDocument();
    expect(starsTitle).toBeInTheDocument();
  });
  test("footer exist with text", () => {
    render(<App />);
    const footerText = screen.getByText("Vilm © 2021");
    expect(footerText).toBeInTheDocument();
  });
  // Снепшот головної сторінки
  test("Project snapshot", () => {
    const mainElement = render(<App />);
    expect(mainElement).toMatchSnapshot();
  });
  test("search element on page", () => {
    render(<App />);
    expect(screen.getByRole("searchbox")).toBeInTheDocument();
  });
});
