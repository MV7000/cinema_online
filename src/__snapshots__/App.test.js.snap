// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Elements on Main Page Project snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="navBarWrapper"
      >
        <a
          class="headerEmblem"
          href="/"
        >
          Vilm
        </a>
        <form
          action="search"
          class="searchWrapper"
        >
          <input
            id="search"
            name="search"
            type="search"
            value=""
          />
        </form>
      </div>
      <div
        class="carousel slide"
      >
        <div
          class="carousel-indicators"
        />
        <div
          class="carousel-inner"
        />
        <a
          class="carousel-control-prev"
          href="#"
          role="button"
          tabindex="0"
        >
          <span
            aria-hidden="true"
            class="carousel-control-prev-icon"
          />
          <span
            class="visually-hidden"
          >
            Previous
          </span>
        </a>
        <a
          class="carousel-control-next"
          href="#"
          role="button"
          tabindex="0"
        >
          <span
            aria-hidden="true"
            class="carousel-control-next-icon"
          />
          <span
            class="visually-hidden"
          >
            Next
          </span>
        </a>
      </div>
      <div
        class="movies_stars_container"
      >
        <div
          class="movies"
        >
          <div
            class="film_title"
          >
            Trending Movies
          </div>
          <div
            class="mov"
          />
        </div>
        <h2
          class="stars_title"
        >
          Trending Stars
        </h2>
        <div
          class="imgWrapper"
        />
      </div>
      <div
        class="footerWrapper"
      >
        <div>
          Vilm © 2021
        </div>
        <p>
          powered by tmdb
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="navBarWrapper"
    >
      <a
        class="headerEmblem"
        href="/"
      >
        Vilm
      </a>
      <form
        action="search"
        class="searchWrapper"
      >
        <input
          id="search"
          name="search"
          type="search"
          value=""
        />
      </form>
    </div>
    <div
      class="carousel slide"
    >
      <div
        class="carousel-indicators"
      />
      <div
        class="carousel-inner"
      />
      <a
        class="carousel-control-prev"
        href="#"
        role="button"
        tabindex="0"
      >
        <span
          aria-hidden="true"
          class="carousel-control-prev-icon"
        />
        <span
          class="visually-hidden"
        >
          Previous
        </span>
      </a>
      <a
        class="carousel-control-next"
        href="#"
        role="button"
        tabindex="0"
      >
        <span
          aria-hidden="true"
          class="carousel-control-next-icon"
        />
        <span
          class="visually-hidden"
        >
          Next
        </span>
      </a>
    </div>
    <div
      class="movies_stars_container"
    >
      <div
        class="movies"
      >
        <div
          class="film_title"
        >
          Trending Movies
        </div>
        <div
          class="mov"
        />
      </div>
      <h2
        class="stars_title"
      >
        Trending Stars
      </h2>
      <div
        class="imgWrapper"
      />
    </div>
    <div
      class="footerWrapper"
    >
      <div>
        Vilm © 2021
      </div>
      <p>
        powered by tmdb
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
